File: RedSheet.DbEntities/Models/vLongTermObjective.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vLongTermObjectives", Schema = "dbo")]
	[Serializable]
	public partial class vLongTermObjective  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int LongTermObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string LongTermObjectiveValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectBackgroundId { get; set; } 

	public vLongTermObjective()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-long-term-objective.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vLongTermObjective {
    constructor(vLongTermObjective?: vLongTermObjective )  {
        let properties = [ "longTermObjectiveId", "longTermObjectiveValue", "projectBackgroundId",];
        for (let property of properties)
            if (vLongTermObjective && vLongTermObjective[property])
                this[property] = vLongTermObjective[property];
    }

	longTermObjectiveId : number =   0 ;

    @required()
    @maxLength(100)
	longTermObjectiveValue : string =   undefined;

    @range(1,2147483647)
	projectBackgroundId : number =   undefined;


}
